website/admin/views/admin_user_rooms.py

```diff
@@ -0,0 +1,51 @@
+from flask import redirect, url_for, flash, render_template
+
+import data_interface
+from admin import admin_site
+from admin.views.admin_user_forms import AddNewRoomForm
+
+
+@admin_site.route('/user/<string:user_id>')
+def user(user_id):
+    rooms = data_interface.get_default_rooms_for_user(user_id)
+    user_info = data_interface.get_user_info(user_id)
+    return render_template("admin/admin_user_home.html", admin=True, rooms=rooms, user_name=user_info["name"])
+
+
+@admin_site.route('/room/add', methods=['POST', 'GET'])
+def add_new_room():
+    form = AddNewRoomForm()
+    if form.validate_on_submit():
+        try:
+            data_interface.add_new_room(name=form.name.data)
+        except Exception:
+            flash("Error", "danger")
+            return render_template("admin/admin_new_room.html", new_room_form=form)
+        flash("New room successfully added!", 'success')
+        return redirect(url_for('.user'))
+    return render_template("admin/admin_new_room.html", new_room_form=form)
+
+
+
+@admin_site.route('/room/<string:room_id>')
+def view_room(room_id):
+    room = data_interface.get_room_info(room_id)
+    all_devices = data_interface.get_user_default_devices()
+    print(all_devices)
+    linked_devices = [d for d in all_devices if d['room_id'] is not None]
```

Comment:
Same as with the other filters: please create functions in a shared file so we can reuse the logic between the internal and external blueprints